Library Genesis.js

```diff
@@ -0,0 +1,245 @@
+{
+  "translatorID": "11614156-f421-4e89-8ce0-a5e69ce3ebed",
+  "label": "Library Genesis",
+  "creator": "Reverend Wicks Cherrycoke",
+  "target": "^https?://(libgen\\.io|lib\\.gen\\.rus\\.ec)/",
+  "minVersion": "3.0",
+  "maxVersion": "",
+  "priority": 100,
+  "inRepository": true,
+  "translatorType": 4,
+  "browserSupport": "gcsi",
+  "lastUpdated": "2016-04-22 22:11:28"
+}
+
+var MIME_TYPES = {
+  'pdf': 'application/pdf',
+  'epub': 'application/epub+zip',
+  'mobi': 'application/x-mobipocket-ebook',
+  'djvu': 'image/vnd.djvu'
+};
+var MD5_REGEX = /md5=([0-9a-fA-F]+)/;
+var AUTHOR_REGEX = /author\s*=\s*{(.*?)}/;
+
+
+function detectWeb(doc, url) {
+  if (url.indexOf('book/index.php') != -1) {
+  return 'book';
+  }
+}
+
+function doWeb(doc, url) {
+  var md5Hash = MD5_REGEX.exec(url)[1];
+
+  // To save some work, we use the provided bibtex file to retrieve the
+  // metadata and use Zotero's built-in bibtex importer
+  var bibtexUrl = "/book/bibtex.php?md5=" + md5Hash;
+  ZU.processDocuments(bibtexUrl, function(bibtexDoc) {
+  var bibtexStr = bibtexDoc.getElementsByTagName("textarea")[0].value
+  var translator = Zotero.loadTranslator('import');
+  translator.setTranslator('9cb70025-a888-4a29-a210-93ec52da40d4');
+  translator.setString(bibtexStr);
+
+  translator.setHandler('itemDone', function(obj, item) {
+    // The bibtex messes up multiple authors, so we set them ourselves
+    var authorStr = AUTHOR_REGEX.exec(bibtexStr)[1];
+    if (authorStr.indexOf(';') != -1) {
+      item.creators = authorStr.split(";").map(function(author) {
+        // Are we dealing with "last, first" formatting?
+        var useCommas = (author.indexOf(",") != -1);
+        return ZU.cleanAuthor(author, "author", useCommas);
+      });
+    } else {
+      item.creators = authorStr.split(",").map(function(author) {
+      return ZU.cleanAuthor(author, "author", false);
+      });
+    }
+
+    // It also messes up multiple ISBNs, so we just pick the first one
+    if (item.ISBN) {
+      [' ', ','].forEach(function(splitChar) {
+      if (item.ISBN.indexOf(splitChar) != -1) {
+        item.ISBN = item.ISBN.split(splitChar)[0];
+      }
+      });
+    }
+
+    // Add the full text attachment
+    var extension = ZU.xpathText(doc, "//tbody/tr[11]/td[4]");
```

Comment:
Two things here:
1. On a general level, I'm a bit concerned about the automatic download here. Some of these files are huge, and I'm not sure how comfortable I am having Zotero auto-download files with sizes like 81MB. @zuphilip @aurimasv  -- thoughts on that one?
2. On a more practical level, the xpath seems very fragile. Wouldn't it be better to test for the `following-sibling` of the node that `contains(text(), "Extension:")` or something along those lines?